src/main.rs

```diff
@@ -135,11 +104,92 @@ h1 {{
                         .unwrap_or(&String::from("")),
                 ),
         );
+    html.to_html_string()
+}
 
-    let mut out: Box<dyn Write> = if let Some(outpath) = matches.get_one::<PathBuf>("output") {
-        Box::new(File::create(outpath)?)
+#[tokio::main]
+async fn service(matches: ArgMatches, names: Vec<String>) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Instead of of starting and crashing the server when something is wrong in the command line arguments, create a configuration object that is fully checked. 
```
struct State {
  names: Vec<String>,
  listen: IPAddress, 
  rng: Random,
}
```
Then call the server using this configuration object.

Reply:
Or pass the necessary thins as distinct arguments. 
Either way is nicer and cleaner than your solution.